docs/developer/data_import.md

```diff
@@ -10,6 +10,55 @@ The general flow of data goes from an external data source, to files uploaded to
 
 The Staging Service acts as a first stop for importing data. This service provides an API to a KBase-hosted file system, where each user has a separate file storage directory under their username. The Narrative front end uses a [REST client](../../kbase-extension/static/kbase/js/api/StagingServiceClient.js) to access that service and manipulate files.
 
+## Staging Service Proxy
+
+We limit staging file uploads to 5GB.
+
+This limit is enforced in the `fileUploadWidget`, but we should also enforce it at the server level as well. To do so, we need to configure the `nginx` proxy server which runs in front of the `staging_service`.
+
+Unfortunately, due to the design of the file upload service, this is accomplished by a limit on the request body, not the file size. In the current implementation of the the fileUploadWidget and Staging Service, the file is transmitted as `multipart/form-data`. This implies that the http request body is larger than the file itself. Therefore, the technique illustrated below adds a small, 1k padding to accommodate the destPath form part as well as the headers for each part.
+
+This section provides an example of how nginx may be configured to support both the request body limit and a custom 413 response.
+
+
+### set the client max body size
+
+Within the server block which handles the reverse proxy or the specific location for the `staging_server` set the `client_max_body_size` to the desired limit.
+
+```nginx
+    client_max_body_size 5000001000;
+```
+
+Note that the body size is 5GB with a 1K bytes padding.
+
+### create a custom response for 413
+
+A custom 413 response can provide more detailed information to the front end to facilitate debugging and better error reports.
+
+For a custom 413 response, first define a location within the server block handling the `staging_server`.
+
+```nginx
```

Comment:
Is this set up in the staging service? I know you were making changes there, too, with @bio-boris .

Reply:
This will come from changes to the nginx proxy. I've included the instructions, at least for what worked for me.

I tested this out by setting the kbase-ui proxy up with a low limit to trigger the 413 for arbitrary file sizes.

Yes, there is a set of changes coming for the staging service too, but the Narrative changes (other than the handling of a custom 413, which is auto-detected) don't depend on them. However, for anyone sending files between 4.4 and 5GB, the current staging service will crash. The upcoming changes address this.